tools/gfx/vulkan/render-vk.cpp

```diff
@@ -3943,6 +3943,118 @@ class VKDevice : public RendererBase
             : public IResourceCommandEncoder
             , public RefObject
         {
+        public:
+        static VkImageLayout translateImageLayout(ResourceState state)
+        {
+            switch (state)
+            {
+            case ResourceState::Undefined:
+                return VK_IMAGE_LAYOUT_UNDEFINED;
+            case ResourceState::PreInitialized:
+                return VK_IMAGE_LAYOUT_PREINITIALIZED;
+            case ResourceState::UnorderedAccess:
+                return VK_IMAGE_LAYOUT_GENERAL;
+            case ResourceState::RenderTarget:
+                return VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+            case ResourceState::DepthRead:
+                return VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL;
+            case ResourceState::DepthWrite:
+                return VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+            case ResourceState::ShaderResource:
+                return VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+            case ResourceState::ResolveDestination:
+            case ResourceState::CopyDestination:
+                return VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+            case ResourceState::ResolveSource:
+            case ResourceState::CopySource:
+                return VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL;
+            case ResourceState::Present:
+                return VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
+            default:
+                assert(!"Unsupported");
+                return VK_IMAGE_LAYOUT_UNDEFINED;
+            }
+        }
+
+        static VkAccessFlagBits calcAccessFlags(ResourceState state)
```

Comment:
This looks good, but we probably need to verify against the spec that we are producing compatible access flags and stage flags.